### PR TITLE
fix(FEC-12347): sanitize before Json parse

### PR DESCRIPTION
**Pull request by lianbenjamin**
**the issue:**
in some cases we might get "invalid" data from BE, such as: entryName: 'invalid \t name'. at some point we are evaluating the data and executing Json.parse, which then fails; such failure is causing the player to not send analytics.

### Description of the Changes

add sanitize function before calling Json.parse to omit invalid json chars.

Solves FEC-12347

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Review by giladna — APPROVED** [review]

File: test/src/common/plugin/plugins-config.spec.js [hunk 1]
```diff
@@ -132,6 +132,14 @@ describe('evaluatePluginsConfig', () => {
     pluginsConfig.kava.should.have.property('playerVersion');
     pluginsConfig.kava.playerVersion.should.equal(__VERSION__);
   });
+
+  it('should sanitize the entryName', () => {
```
Comment (giladna) [inline test/src/common/plugin/plugins-config.spec.js:136]: @lianbenjamin  what does this method is doing?

File: src/common/plugins/plugins-config.js [hunk 2]
```diff
@@ -141,6 +143,30 @@ function getEncodedReferrer(): string {
...
 }
 
+/**
+ * @private
+ * @param {string} text - the string to sanitize
+ * @returns {string} - the sanitized string
+ * @private
+ */
+function _sanitize(text: string): string {
+  if (!text) return;
```
Comment (giladna) [inline src/common/plugins/plugins-config.js:153]: @lianbenjamin  this line fail in flow 
https://app.travis-ci.com/github/kaltura/kaltura-player-js/jobs/583634774

File: src/common/plugins/plugins-config.js [hunk 3]
```diff
@@ -149,6 +175,7 @@ function getEncodedReferrer(): string {
  */
 function _formatConfigString(config: string): Object {
   let configObj;
+  config = _sanitize(config);
```
Comment (giladna) [inline src/common/plugins/plugins-config.js:178]: @lianbenjamin seems need to check here if config is not empty then you do not need the validation there 
and avoid parsing empty stuff

**Review by giladna — CHANGES_REQUESTED** [review]

File: test/src/common/plugin/plugins-config.spec.js (hunk 1 above)
Comment (lianbenjamin) [reply test/src/common/plugin/plugins-config.spec.js:136]: it's a test- I pass an invalid payload and make sure it's sanitized and exist on the final object.

File: src/common/plugins/plugins-config.js (hunk 3 above)
Comment (lianbenjamin) [reply src/common/plugins/plugins-config.js:178]: I check inside _sanitize method

**Review by giladna — APPROVED** [review]

**Pull request merged**